Lib/test/test_netrc.py

```diff
@@ -23,6 +24,22 @@ def make_nrc(self, test_data):
             os.unlink(temp_filename)
         return nrc
 
+    def make_nrc_file(self, test_data, filename=None):
+        test_data = textwrap.dedent(test_data)
+        temp_data = StringIO(test_data)
+        try:
+            nrc = netrc.netrc(temp_data, filename)
+        finally:
+            pass
+        return nrc
+
+    def test_nrc_filename(self):
+        with self.assertRaisesRegex(netrc.NetrcParseError, "bad toplevel token 'spam' \\(eggs, line 1\\)"):
+            self.make_nrc_file("spam", "eggs")
+
+        with self.assertRaisesRegex(netrc.NetrcParseError, "bad toplevel token 'spam' \\(<_io.StringIO object at 0x[0-9a-fA-F]+>, line 1\\)"):
```

Comment:
It can be `_pyio.StringIO` in other implementations. It is better to be more lenient: `<.*StringIO.*>`.